server/game/cards/03_TWI/leaders/GeneralGrievousGeneralOfTheDroidArmies.ts

```diff
@@ -0,0 +1,45 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { KeywordName, Trait, WildcardCardType } from '../../../core/Constants';
+
+export default class GeneralGrievousGeneralOfTheDroidArmies extends LeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '2872203891',
+            internalName: 'general-grievous#general-of-the-droid-armies',
+        };
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addActionAbility({
+            title: 'Give a Droid unit Sentinel for this phase',
+            cost: [AbilityHelper.costs.exhaustSelf()],
+            targetResolver: {
+                cardTypeFilter: WildcardCardType.Unit,
+                cardCondition: (card) => card.isUnit() && card.hasSomeTrait(Trait.Droid),
```

Comment:
We can remove the `isUnit()` here since we have the `cardTypeFilter` above. Same for the deployed ability